utils/wasm-gen/src/generator/syscalls/invocator.rs

```diff
@@ -315,6 +332,49 @@ impl<'a, 'b> SyscallsInvocator<'a, 'b> {
         };
         instructions.append(&mut result_processing);
 
+        if let Some(waiting_probability) = self
```

Comment:
can we move all this logic to a separate method and add a explanatory doc to it?